docs/user/agent/authentication-authorization.md

```diff
@@ -0,0 +1,160 @@
+4. If the token in the secret is older than 24 hours when creating a node ISO, the installer generates new tokens. These new tokens are saved in the asset store, and the secret in the cluster is updated with the new token values.
+5. If the token in the secret is not older than 24 hours, the installer retrieves the existing token from the cluster, updates the asset store with the retrieved token, and uses the retrieved token for the new node ISO.
+6. When a node is booted using the Day 2 `nodes.iso` to add worker nodes, the `agent-auth-token-status.service` systemd service will check if the authentication tokens have expired. If the tokens are expired, the following message will be displayed:
+
+```
+The authentication token has expired. Please generate a new ISO using the "oc adm node-image create" command, then reboot the node.
+```
+
+This ensures that the user is notified of expired tokens and can take appropriate action to generate a new ISO.
+
+```mermaid
+flowchart TD
+    A[Check if openshift-config/agent-auth-token exists] -->|No| B[Create secret with 3 tokens and public key]
+    A -->|Yes| C[Retrieve tokens from existing secret]
+    B --> D[Generate new tokens if secret does not exist]
+    C --> D[Check if token is older than 24 hours]
+    D -->|Yes| E[Generate new tokens, update asset store, and secret]
+    D -->|No| F[Retrieve token, update asset store, use token for ISO]
+    F --> G[Node boots and checks token expiry]
+    G -->|Expired| H[Display message: 'Authentication token expired. Generate a new ISO.']
+    G -->|Not expired| I[Proceed with installation using retrieved token]
+```
+## Authorization
+
+To implement authorization for ABI, a new security definition is added in the `swagger.yaml` file in AS. The new security definition is as follows:
+
+```yaml
+watcherAuth:
+  type: apiKey
+  in: header
+  name: Watcher-Authorization
+```
+
+The required endpoints are annotated with this security definition. For example:
+
+```yaml
+/v2/clusters:
+  get:
+    tags:
+      - installer
+    security:
+      - userAuth: [admin, read-only-admin, user]
+      - watcherAuth: []
+    description: Retrieves the list of OpenShift clusters.
+    operationId: v2ListClusters
+```
+
+### Authorization Handler
+
+The Assisted Service API includes an authorization handler implemented in the `pkg/auth/agent_local_authz_handler.go` file. The `AgentLocalAuthzHandler`  is responsible for verifying that the `auth_scheme` in the token matches the required `authScheme` for the endpoint. The authorization logic works as follows:
+
+1. The `auth_scheme` claim in the JWT token is extracted.
+2. If the claim is malformed or missing, an error is returned.
+3. The `auth_scheme` claim embedded in the token specifies the type of authorization (e.g., `watcherAuth`) and is compared with the `authScheme` expected by the endpoint. This ensures that the token matches the intended purpose of the request.  
```

Comment:
Is /v2/clusters above the only endpoint that supports watcherAuth?

Reply:
No, the full details are in the swagger.yaml here https://github.com/openshift/assisted-service/blob/master/swagger.yaml#L169